### PR TITLE
Future - Replace syntax highlighter in lib/components

### PR DESCRIPTION
**Pull request by okonet**
Issue: https://github.com/storybookjs/storybook/issues/18090

## What I did

After the investigation, I found out that the issue was related to https://github.com/react-syntax-highlighter/react-syntax-highlighter and specifically to global PrismJS instance that would replace the already rendered contents of `code` blocks.

This PR replaces the rendering engine `SyntaxHighlighter` component to use https://github.com/FormidableLabs/prism-react-renderer.

## How to test

Check out https://github.com/storybookjs/storybook/issues/18090

## Breaking changes

Technically it's a breaking change since the external API of `SyntaxHighlighter` has been changed. Check out https://github.com/storybookjs/storybook/pull/18146/files#diff-067280cfd28572f546ce2152e8b22d2b2736bf7cb3dc354e908d3e8bd5208b5e. Since we consider this component as an internal component, I think it's still fine to release this under `6.5` and adding this to the release notes. 

Other than that, I preserved the current API as much as possible as well as underlying class names to some extent. So if someone relies on class names like `token tag` etc. for custom styling it should still work. 

### Before
CleanShot 2022-05-05 at 16 51 19@2x

### After
CleanShot 2022-05-05 at 16 50 48@2x

## Bundle size (bytes)

Running `cd examples/react-ts && yarn && yarn build-storybook` yields

`next`:  `373192`
`replace-syntax-highlighter`: `333416`
Diff: `-39776`

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -7517,11 +7517,11 @@ __metadata:
     overlayscrollbars: ^1.13.1
     polished: ^4.2.2
     prettier: ">=2.2.1 <=2.3.0"
+    prism-react-renderer: ^1.3.1
     prop-types: ^15.7.2
     qs: ^6.10.0
     react-colorful: ^5.1.2
     react-popper-tooltip: ^3.1.1
-    react-syntax-highlighter: ^15.4.5
```
Comment (okonet) [inline yarn.lock:7524]: Interestingly this one didn't cause a bigger change here. Is it sill being used by other components?

**Comment by shilman** [comment]
@okonet build is failing
> ERR! FAILED (ts) : src/blocks/mdx.tsx:44:7 - error TS2322: Type 'string' is not assignable to type 'Language'
>       language={(language && language[1]) || 'plaintext'}

**Comment by ndelangen** [comment]
@shilman This doesn't look like it's on anyone's plate right now. What do we do here?

**Comment by ndelangen** [comment]
@okonet I migrated this PR from being based on `next` but targeting `future/base` to be based on `future/base` I'm real sorry the commit history was lost in this process 😞

**Pull request closed without merging**